### ISSUE TITLE
Can't able to get new uri of photo after moving it to other folder. [Android]

### ISSUE THREAD

**Issue by CavidM** [expo/expo#22817]
### Minimal reproducible example

https://snack.expo.dev/@cavidm/media-library-move-asset?platform=android

### It happens only on android
**Steps**:
1. Pick or take a photo with expo-image-picker
2. Save the photo by using MediaLibrary.createAssetAsync(imageUri). So you have an Asset
3. Move the photo to another folder with MediaLibrary.createAlbumAsync or MediaLibrary.addAssetsToAlbumAsync

**After that**:
1. You want to see your photo. 
2. But your photo uri or Asset points to the old folder. And the image doesn't exists in that folder.
3. So you end up by loosing the image.
4. Wanted to get new uri for the image with MediaLibrary.getAssetInfoAsync(asset). It returns null.

This flow should have the same result as in ios. Asset should point to the new place of media.

### Environment

expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.4
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.14.0 - /usr/local/bin/node
      Yarn: 1.22.19 - /usr/local/bin/yarn
      npm: 9.3.1 - /usr/local/bin/npm
    Managers:
      CocoaPods: 1.12.1 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
    IDEs:
      Android Studio: 2022.2 AI-222.4459.24.2221.10121639
      Xcode: 14.3.1/14E300c - /usr/bin/xcodebuild
    npmPackages:
      expo: ~48.0.18 => 48.0.19 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.71.8 => 0.71.8 
    npmGlobalPackages:
      expo-cli: 6.3.7
    Expo Workflow: managed

**Comment by alanjhughes** [comment]
Hi @CavidM - You are creating an asset, moving it without copying, and then using your original reference to it in `getAssetInfoAsync`. This will be null as it has been moved. You will need to fetch the asset from the album after moving it or not pass false to `createAlbumAsync` so the asset is copied.

**Comment by alanjhughes** [comment]
You can get the assets from a folder like this
```ts
 const album = await MediaLibrary.getAlbumAsync("MyAlbum");
 const assets = await MediaLibrary.getAssetsAsync({ album: album });
```

**Comment by alanjhughes** [comment]
I've modified your snack with a working example https://snack.expo.dev/dswBvUPw_

**Comment by CavidM** [comment]
Hi @alanjhughes . Thank you for your response.

Here is why I can't use the solution you provided.
1. In short, I want to have the same result as in ios. (I move the asset and I have the right asset reference with correct `uri`)
2. I can't remove "`false`" flag, because i want to move the `asset`, not copy. (When I copy the `asset` it still points to the old folder but I need to have the new `uri`)
3. I can't use `MediaLibrary.getAssetsAsync`. It doesn't' work for this case. It returns all `assets` with pagination with different `ids`. 
- But I found a way with `MediaLibrary.getAssetsAsync`. I tried with different options: {`after: lastAssetId`} {`createdAfter: someTimeAgo()`}. But none of these work. Looks like it doesnt' work whenever i add "`after`" or "`createdAfter`" props.
I see my `asset` was there, but i can't able to get it. In better case I need to get assets created a min ago or a few seconds ago. (Because these things happens immediately, creating and moving) or I can get assets which ids comes after my asset.id before moved.

For now I'm just checking if the platform is ios, it creates a new asset and moves it otherwise it does nothing but only creating new asset without moving it.
